src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1724,6 +1734,48 @@ private function registerCacheConfiguration(array $config, ContainerBuilder $con
         }
     }
 
+    private function registerHttpClientConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader)
+    {
+        if (!class_exists(HttpClient::class)) {
+            throw new LogicException('HttpClient support cannot be enabled as the component is not installed. Try running "composer require symfony/http-client".');
+        }
+
+        $loader->load('http_client.xml');
+
+        $merger = new class() {
+            use HttpClientTrait;
+
+            public function merge(array $options, array $defaultOptions)
+            {
+                try {
+                    [, $options] = $this->prepareRequest(null, null, $options, $defaultOptions);
+
+                    return $options;
+                } catch (TransportExceptionInterface $e) {
+                    throw new InvalidArgumentException($e->getMessage(), 0, $e);
+                }
+            }
+        };
+
+        $defaultOptions = $merger->merge($config['default_options'] ?? [], []);
+        $container->getDefinition('http_client')->setArguments([$defaultOptions, $config['max_host_connections']]);
+
+        foreach ($config['clients'] as $name => $clientConfig) {
+            $options = $merger->merge($clientConfig['default_options'] ?? [], $defaultOptions);
+
+            $container->register($name, HttpClientInterface::class)
+                ->setFactory([HttpClient::class, 'create'])
+                ->setArguments([$options, $clientConfig['max_host_connections']]);
```

Comment:
If I don't miss anything, this cannot be set when using the XML config  format.

Reply:
not sure what you missed, I added it in the fixtures and it worked - or did you mean something else? (see fixtures)